source/uwp/Renderer/lib/AdaptiveTextBlockRenderer.cpp

```diff
@@ -89,4 +115,52 @@ namespace AdaptiveNamespace
             jsonObject, elementParserRegistration, actionParserRegistration, adaptiveWarnings, element);
     }
     CATCH_RETURN;
+
+    HRESULT AdaptiveTextBlockRenderer::GetHeadingLevelFromContext(ABI::AdaptiveNamespace::IAdaptiveRenderContext* renderContext,
+                                                                  ABI::Windows::UI::Xaml::Automation::Peers::AutomationHeadingLevel* headingLevel)
+    {
+        ComPtr<IAdaptiveHostConfig> hostConfig;
+        RETURN_IF_FAILED(renderContext->get_HostConfig(&hostConfig));
+
+        ComPtr<IAdaptiveHeadingsConfig> headingConfig;
+        RETURN_IF_FAILED(hostConfig->get_Headings(&headingConfig));
+
+        unsigned int levelInt;
+        RETURN_IF_FAILED(headingConfig->get_Level(&levelInt));
+
+        switch (levelInt)
+        {
+        case 0:
+        case 1:
+            *headingLevel = AutomationHeadingLevel_Level1;
+            break;
+        case 2:
+            *headingLevel = AutomationHeadingLevel_Level2;
+            break;
+        case 3:
+            *headingLevel = AutomationHeadingLevel_Level3;
+            break;
+        case 4:
+            *headingLevel = AutomationHeadingLevel_Level4;
+            break;
+        case 5:
+            *headingLevel = AutomationHeadingLevel_Level5;
+            break;
+        case 6:
+            *headingLevel = AutomationHeadingLevel_Level6;
+            break;
+        case 7:
+            *headingLevel = AutomationHeadingLevel_Level7;
+            break;
+        case 8:
+            *headingLevel = AutomationHeadingLevel_Level8;
+            break;
+        case 9:
+        default:
+            *headingLevel = AutomationHeadingLevel_Level9;
```

Comment:
nit: from the discussion, the reasonable default value seems to be level 2.

Reply:
Hmm.... So the default coming from the host config definition will be 2, so if they don't put anything that's what they'll get. But this is the scenario where someone has explicitly put like 11 or something in their host config. My thinking is if they put a really big number in we should use the biggest one we have available.